fix(block, block-section): improve a11y

jcfranco opened the pull request:
**Related Issue:** #5565 

## Summary

Updated HTML to improve a11y.

**Note**: this removes references to outdated `TEXT` constants, which were used before translations were built-in.

driskull reviewed (approved):
Nice improvements! 🦾

geospatialem reviewed (commented):
Know this isn't part of the PR, but the messages are repeating the collapse/expand twice. Would it be possible to remove the `toggleLabel`? The `aria-expanded` should provide context, but it looks like context is provided in the `title`, which might be the duplication presented in AT. 😢

jcfranco commented:
@geospatialem I made the following changes according to https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/:

* removed `aria-label` since we have the same content as visible text
* removed `aria-expanded` on the content area since it's only needed in the toggle button (changed container back to `section`)
* ensured toggle button has `button` role and `aria-expanded`
* associated the content with the toggle button w/ `aria-labelledby`
* added `aria-hidden` to the switch since it is used as presentation (toggle button has all the relevant state)

Could you test again? 🙇‍♂️

geospatialem commented on packages/calcite-components/src/components/block-section/block-section.tsx:
```diff
@@ -259,13 +259,7 @@ export class BlockSection implements LocalizedComponent, T9nComponent, LoadableC
-          hidden={!open}
-          id={regionId}
-        >
+        <section aria-labelledby={buttonId} class={CSS.content} hidden={!open} id={regionId}>
```
Not to the full context of the PR, but adding the following to the `section` provide additional context with JAWS when the section's content contains text and is expanded:
```ts
aria-live="polite"
role="region"
```

geospatialem replied:
Per our Teams convo, and Franco's sleuthing, [`section` inherits a region `role`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section#technical_summary), so the `role` is not needed in addition to `aria-live`.

geospatialem commented on packages/calcite-components/src/components/block/block.tsx:
```diff
@@ -303,7 +303,6 @@ export class Block
           <button
             aria-controls={regionId}
             aria-expanded={collapsible ? toAriaBoolean(open) : null}
-            aria-label={toggleLabel}
             class={CSS.toggle}
             id={buttonId}
             onClick={this.onHeaderClick}
```
It seems the `title` attribute is conflicting with AT, but if we add an `aria-describedby` and an unique `id` using `guid` to the `header`, similar to the `buttonId` and `regionId`, we could provide both contexts.

Something like:

```ts
const headerId = `${guid}-header`;

...

<header class={CSS.header} id={headerId}>

...

<button
...
aria-describedby={headerId}
...
>


```

geospatialem reviewed (commented):
A few small tweaks for a11y, and this should be good to go!

jcfranco commented:
@geospatialem Belated update, I rolled back a few changes to where the AT interpretation was better and also restructured the `block-section` internals to avoid nested interactive elements. 

I think you beat me to the 💥🥊 with https://github.com/Esri/calcite-design-system/pull/7652, but we can make further adjustments if needed.
